tests/test_quixstreams/test_dataframe/test_joins/test_lookup_quix_config.py

```diff
@@ -0,0 +1,1117 @@
+        assert result == version2
+        assert config.version == version2
+        assert config.previous_version == version1
+        assert config.next_version == version3
+
+        # Between second and third
+        result = config.find_valid_version(2500)
+        assert result == version2
+
+        # At third version
+        result = config.find_valid_version(3000)
+        assert result == version3
+        assert config.version == version3
+        assert config.previous_version == version2
+        assert config.next_version is None
+
+        # After third version
+        result = config.find_valid_version(4000)
+        assert result == version3
+
+    def test_find_valid_version_cache_invalidation_forward_time(self):
+        """Test cache invalidation when time moves forward and next version becomes valid."""
+        version1 = create_configuration_version(version=1, valid_from=1000.0)
+        version2 = create_configuration_version(version=2, valid_from=2000.0)
+
+        config = Configuration(versions={1: version1, 2: version2})
+
+        # First call at timestamp 1500 - should cache version1 as current, version2 as next
+        result = config.find_valid_version(1500)
+        assert result == version1
+        assert config.version == version1
+        assert config.next_version == version2
+
+        # Second call at timestamp 2500 - next version should now be current
+        result = config.find_valid_version(2500)
+        assert result == version2
+        assert config.version == version2
+        assert config.previous_version == version1
+        assert config.next_version is None
+
+    def test_find_valid_version_out_of_order_messages(self):
+        """Test handling out-of-order messages (timestamp going backwards)."""
+        version1 = create_configuration_version(version=1, valid_from=1000.0)
+        version2 = create_configuration_version(version=2, valid_from=2000.0)
+
+        config = Configuration(versions={1: version1, 2: version2})
+
+        # First call at timestamp 2500 - should get version2
+        result = config.find_valid_version(2500)
+        assert result == version2
+        assert config.version == version2
+        assert config.previous_version == version1
+        assert config.next_version is None
+
+        # Second call at timestamp 1500 (backwards in time) - should get version1
+        result = config.find_valid_version(1500)
+        assert result == version1
+        assert config.version == version2
+        assert config.previous_version == version1
+        assert config.next_version is None
+
+    def test_find_valid_version_mixed_timestamp_and_no_timestamp(self):
+        """Test finding valid version with mix of timestamped and non-timestamped versions."""
+        version1 = create_configuration_version(version=1, valid_from=1000.0)
+        version2 = create_configuration_version(
+            version=2, valid_from=0.0
+        )  # No timestamp
```

Comment:
I believe it will be more readable without a comment. It's clear enough from the docstring context.
```suggestion
        version2 = create_configuration_version(version=2, valid_from=0.0)
```